packages/ui-components/src/components/form/RadioGroup.vue

```diff
@@ -0,0 +1,53 @@
+<template>
+  <div>
+    <div class="sr-only">{{ label }}</div>
+    <div class="flex flex-col gap-2">
+      <div
+        v-for="(option, index) in options"
+        :key="index"
+        class="flex items-center gap-3"
+      >
+        <input
+          :id="`option-${index}`"
+          v-model="selected"
+          type="radio"
+          :value="option"
+          class="cursor-pointer h-3 w-3 bg-foundation border-foundation ring-[1.5px] ring-primary checked:ring-primary-focus focus:ring-offset-foundation focus:outline-primary-focus"
+        />
+        <label
+          class="flex items-center gap-1.5 text-sm text-foreground cursor-pointer"
+          :for="`option-${index}`"
+        >
+          <Component :is="option.icon" v-if="option.icon" class="h-9 w-9" />
+          {{ option.title }}
+        </label>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, watchEffect } from 'vue'
+import type { ConcreteComponent } from 'vue'
+
+interface Option {
+  title: string
+  icon?: ConcreteComponent
+}
+
+const props = defineProps<{
+  label: string
+  options: Option[]
+}>()
+
+const emit = defineEmits(['update:selected'])
+
+const selected = ref<Option | null>(props.options[0] || null)
+
+watchEffect(() => {
```

Comment:
Why `watchEffect` instead of just `watch`? I've never had a need to use it and it's essentially a watcher that triggers whenever any of the refs used inside of its callback are updated